lib/x11utils.pm

```diff
@@ -223,6 +223,8 @@ sub handle_login {
     assert_screen 'displaymanager-password-prompt';
     type_password;
     send_key 'ret';
+    assert_screen([qw(generic-desktop gnome-activities opensuse-welcome-gnome40-activities opensuse-welcome)], 60);
```

Comment:
Using `opensuse-welcome` here might lead to failures if this happens in modules which expect `generic-desktop` only, which is probably most of them.

Reply:
Remove `opensuse-welcome` here will lead to the first_boot fail here: https://openqa.opensuse.org/tests/1837384#step/first_boot/5

Reply:
Yes, I'm not sure how to deal with that properly either TBH. It would be ideal if `handle_login` could be told which tags are acceptable, but in that case it runs before the `opensuse_welcome` module so it's not possible.

The easiest option is indeed to just always accept `opensuse_welcome` needles here and just delay the failure that way...

Reply:
I totally agree with you. 
So, I just updated the code to add the tag `opensuse-welcome`.